Quote XAuthLocation before use

riverar opened the pull request:
`XAuthLocation` is supported in ssh configuration data and specifies the full pathname to an xauth binary (e.g. `"C:\Program Files\VcXsrv\xauth.exe"`). On Windows, this is passed to the command interpreter (via _wsystem), which requires paths with spaces to be escaped or quoted. Use of a path with spaces will fail some X11 scenarios.

This PR simply adds double-quotes `" "` around `XAuthLocation` paths right before they're used in X11 scenarios on Windows.

The following constraints were assumed:
- Zero impact to non-Windows platforms (ifdef guards in place)
- Minimize diff to maintain attractiveness for upstream consideration

(fixes PowerShell/Win32-OpenSSH#1563, PowerShell/Win32-OpenSSH#1593)

riverar committed:
Quote XAuthLocation before use

riverar commented:
Test note: If https://github.com/PowerShell/openssh-portable/pull/441 is not merged, `TMPDIR` is required to be defined.

bagajjal commented on clientloop.c:
```diff
@@ -357,7 +357,7 @@ client_x11_get_proto(struct ssh *ssh, const char *display,
 					x11_timeout_real = UINT_MAX;
 				}
-				xasprintf(&cmd, "%s -f %s generate %s %s "
+				xasprintf(&cmd, "\"%s\" -f %s generate %s %s "
```
Please have this only for windows although it doesn't make any regression to unix code.

I would like to reproduce the problem myself so please provide the steps.

riverar replied:
@bagajjal Done.

riverar commented:
Repro steps:
1. Edit ssh config, include `XAuthLocation="C:\Program Files\VcXsrv\xauth.exe"` or another path with spaces.
2. If #441 not merged yet, set environment variable TMPDIR to writable location (e.g. C:\Temp)
3. Set up .Xauthority:
    ```
    "C:\\Program Files\\VcXsrv\\xauth.exe" generate 127.0.0.1:0.0 . trusted
    ```
4. `ssh -X -vvv user@host`
5. After authentication, observe:
    ```
    debug2: client_x11_get_proto: xauth command: "C:\\Program Files\\VcXsrv\\xauth.exe" -f ...
    ...
    debug2: x11_get_proto: "C:\\Program Files\\VcXsrv\\xauth.exe" -f ...
    ...
    Warning: untrusted X11 forwarding setup failed: xauth key data not generated
    ```

halflumi commented:
What's the status of this PR?

bagajjal committed:
Update clientloop.c

Minor change to limit the windows specific change.
This results in merge conflicts if the upstream make changes in this code path and also to rerun the CI tests for this PR.

bagajjal committed:
Update clientloop.c

minimize the changes.
